src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+            String detailDescription = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.DETAIL_DESCRIPTION);
+            String bankCode = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.BANK_CODE);
+            String documentDescription = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.DOCUMENT_DESCRIPTION);
+
+            advanceDepositDocument = (AdvanceDepositDocument) documentService.getNewDocument(KFSConstants.FinancialDocumentTypeCodes.ADVANCE_DEPOSIT);
+            advanceDepositDocument.getDocumentHeader().setDocumentDescription(documentDescription);
+            advanceDepositDocument.setCampusLocationCode(CuFPConstants.ADVACE_DEPOSIT_DEFAULT_CAMPUS_CODE);
+            advanceDepositDocument.setDepositDate(dateTimeService.convertToSqlDate(transaction.getBankTimestamp()));
+
+            createSourceAccountingLine(transaction, advanceDepositDocument);
+
+            AdvanceDepositDetail advanceDepositDetail = new AdvanceDepositDetail();
+            advanceDepositDetail.setDocumentNumber(advanceDepositDocument.getDocumentNumber());
+            advanceDepositDetail.setFinancialDocumentAdvanceDepositDate(dateTimeService.convertToSqlDate(transaction.getLoadTimestamp()));
+            advanceDepositDetail.setFinancialDocumentAdvanceDepositAmount(transaction.getTransactionAmount());
+            advanceDepositDetail.setFinancialDocumentAdvanceDepositReferenceNumber(detailReferenceNumber);
+            advanceDepositDetail.setFinancialDocumentAdvanceDepositDescription(detailDescription);
+            advanceDepositDetail.setFinancialDocumentBankCode(bankCode);
+            Bank bank = bankService.getByPrimaryId(bankCode);
+            advanceDepositDetail.setBank(bank);
+
+            advanceDepositDocument.addAdvanceDeposit(advanceDepositDetail);
+        } catch (WorkflowException | ParseException e) {
+            LOG.error("Error creating advance deposit documents: " + e.getMessage(), e);
+            throw new RuntimeException("Error creating advance deposit documents: " + e.getMessage(), e);
+        }
+
+        return advanceDepositDocument;
+    }
+
+    private void createNotes(AchIncomeTransaction transaction, AdvanceDepositDocument document) {
+
+        String attachmentFile = attachmentsDirectory + "/" + CuFPConstants.ADVANCE_DEPOSIT_NOTE_FILE_PREFIX
+                + document.getDocumentNumber() + "_" + new SimpleDateFormat("yyyyMMdd_HHmmss").format(dateTimeService.getCurrentDate()) + ".txt";
+        BufferedWriter os = null;
+
+        try {
+            os = new BufferedWriter(new FileWriter(attachmentFile));
```

Comment:
Is it strictly necessary for the ACH notes to be written to an actual file before being converted into a note attachment file? Would it make more sense to use a byte array (and byte array streams) instead?